group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3038,6 +3042,59 @@ private void replaceMember(
         ));
     }
 
+    /**
+     * Maybe delete the resolved regular expression associated to the provided member if
+     * it was the last subscribed member to it.
+     *
+     * @param records   The record accumulator.
+     * @param group     The group.
+     * @param member    The member removed from the group.
+     */
+    private void maybeDeleteResolvedRegularExpression(
+        List<CoordinatorRecord> records,
+        ConsumerGroup group,
+        ConsumerGroupMember member
+    ) {
+        if (isNotEmpty(member.subscribedTopicRegex()) && group.numSubscribedMembers(member.subscribedTopicRegex()) == 1) {
+            records.add(newConsumerGroupRegularExpressionTombstone(
+                group.groupId(),
+                member.subscribedTopicRegex()
+            ));
+        }
+    }
+
+    /**
+     * Maybe delete the resolved regular expressions associated with the provided members
+     * if they were the last ones subscribed to them.
+     *
+     * @param records   The record accumulator.
+     * @param group     The group.
+     * @param members   The member removed from the group.
+     * @return The set of deleted regular expressions.
+     */
+    private Set<String> maybeDeleteResolvedRegularExpressions(
```

Comment:
should we add a test for this? or include in another test where a single member is subscribed to topic "foo" as both a topic name and as a regex, i.e. "foo" and "foo*"

Reply:
this is tested by `testResolvedRegularExpressionsRemovedWhenConsumerMembersRemovedByAdminApi`. i am not sure to follow your suggestion about having a member subscribed to both `foo` and `foo*`. this code does not look at topic names at all. could you please elaborate?

Reply:
ah, got it. let me extend that test.